chsdi/views/files.py

```diff
@@ -128,25 +129,40 @@ def options_files_collection(self):
             'Access-Control-Allow-Credentials': 'true'})
         return ''
 
-    def _save_to_s3(self, data, mime, update=False, compress=False):
+    def _gzip_data(self, data):
+
+        out = None
+        infile = StringIO.StringIO()
+        try:
+            gzip_file = gzip.GzipFile(fileobj=infile, mode='w', compresslevel=5)
+            gzip_file.write(data)
+            gzip_file.close()
+            infile.seek(0)
+            out = infile.getvalue()
+        except:
+            out = None
+        finally:
+            infile.close()
+        return out
+
+    def _save_to_s3(self, data, mime, update=False, compress=True):
         ziped_data = None
+        content_encoding = None
+        headers = {}
         if compress and mime == 'application/vnd.google-earth.kml+xml':
-            tmp = StringIO.StringIO()
-            zf = zipfile.ZipFile(tmp, mode='w', compression=zipfile.ZIP_DEFLATED, )
-            try:
-                zf.writestr('doc.kml', data)
-                mime = 'application/vnd.google-earth.kmz'
-            finally:
-                zf.close()
-            ziped_data = tmp.getvalue()
+            ziped_data = self._gzip_data(data)
+            content_encoding = 'gzip'
+            headers['Content-Encoding'] = 'gzip'
 
         if not update:
-            if ziped_data is not None:
+            if content_encoding == 'gzip' and ziped_data is not None:
                 data = ziped_data
             try:
-                k = Key(self.bucket)
+                k = Key(bucket=self.bucket)
                 k.key = self.file_id
                 k.set_metadata('Content-Type', mime)
+                k.content_type = mime
+                k.content_encoding = content_encoding
```

Comment:
1/ If a _metadata_ is `None`it is not set (won't be sent to the client)
2/ Because it is not necessary